custom/paystack/infrastructure/event/src/main/java/com/paystack/fineract/infrastructure/config/PaystackKafkaAsyncConfiguration.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.paystack.fineract.infrastructure.config;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.scheduling.annotation.EnableAsync;
+
+/**
+ * Configuration for asynchronous Kafka notification processing.
+ */
+@AutoConfiguration
+@EnableAsync
+@ComponentScan(basePackages = "com.paystack.fineract.infrastructure.event")
+@ConditionalOnProperty(value = "fineract.events.external.producer.kafka.enabled", havingValue = "true")
+public class PaystackKafkaAsyncConfiguration {
+
+    @Value("${fineract.events.external.producer.kafka.async.core-pool-size:5}")
+    private int corePoolSize;
+
+    @Value("${fineract.events.external.producer.kafka.async.max-pool-size:20}")
+    private int maxPoolSize;
+
```

Comment:
I think the root should be paystack.events.. to avoid confusion with default fineract kafka properties.